### PR TITLE
Add to #16916

### PR DESCRIPTION
**Pull request by lavalamp**
We thought maybe it'd be better if this delay was longer. (supersedes #16916)

### CODE REVIEW

File: pkg/storage/cacher_test.go [hunk 1]
```diff
@@ -223,6 +223,32 @@ func TestWatch(t *testing.T) {
...
+	defer watcher.Stop()
+
+	// Create a second watcher that will be reading result.
+	readingWatcher, err := cacher.WatchList(context.TODO(), "pods/ns", 1, storage.Everything)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	defer readingWatcher.Stop()
+
+	for i := 1; i <= 22; i++ {
```
Comment (krousey) [inline pkg/storage/cacher_test.go]: What are these magic numbers? Is it that the cacheWatchers have input and result channels with buffer size 10? so enough to block both?

Comment (wojtek-t) [inline pkg/storage/cacher_test.go]: Yes - exactly.
Both input and output have size 10 and there can we 1 "in flight" during the translation.
So with 21 the tests was still passing without the fix.

**Comment by wojtek-t** [comment]
LGTM (although - I wrote most of this code, so I shouldn't review it) :-)

**Comment by krousey** [comment]
LGTmewwwww. We should really try to clean this up, but this is a fine patch. I would add a comment for the magic numbers.

**Comment by lavalamp** [comment]
Adding krousey's LGTM.

**Comment by wojtek-t** [comment]
@k8s-bot test this please

**Pull request merged**